torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -76,6 +76,36 @@ bool ShouldSyncIrValue(const torch::lazy::Value& ir_value) {
   return ir_value->op() != xla_not_supported;
 }
 
+XLAGraphExecutor::ComputationCache* CreateComputationCache() {
+  static const size_t kMaxCacheSize =
+      runtime::sys_util::GetEnvInt("XLA_COMPILATION_CACHE_SIZE", 1024);
+  static const bool readonlyPersistentCache =
+      runtime::sys_util::GetEnvBool("XLA_PERSISTENT_CACHE_RO", false);
+  static std::string persistentCacheDir =
+      runtime::sys_util::GetEnvString("XLA_PERSISTENT_CACHE_PATH", "");
+  if (!persistentCacheDir.empty()) {
+    auto serialize_fn = [](auto& computation) -> std::string {
+      return runtime::GetComputationClient()->SerializeComputation(
+          computation->computation);
+    };
+    auto deserialize_fn = [](auto& serialization)
+        -> std::shared_ptr<XLAGraphExecutor::CachedComputation> {
+      auto computation =
+          runtime::GetComputationClient()->DeserializeComputation(
+              serialization);
+      if (!computation) return nullptr;
+      bool is_sharded = bridge::GetDefaultDevice()->toString() ==
+                        GetVirtualDevice().toString();
```

Comment:
why do we need to check current default device type?

Reply:
I'm a bit outdated on how to check if we're using SPMD mode in the C++ layer - I think this should be the equivalent of what used to be `ShardingUtil::UseVirtualDevice`, but let me know if I'm mistaken.

Reply:
Turns out we still have a utility for this! Thanks @JackCaoG 

https://github.com/pytorch/xla/blob/6e661304e270c6a1a864700e04d5d44235c43f3b/torch_xla/csrc/device.cpp#L101-L105